forms/account-freemarker/src/main/java/org/keycloak/account/freemarker/FreeMarkerAccountProvider.java

```diff
@@ -197,7 +197,11 @@ public Response createResponse(AccountPages page) {
             String result = freeMarker.processTemplate(attributes, Templates.getTemplate(page), theme);
             Response.ResponseBuilder builder = Response.status(status).type(MediaType.TEXT_HTML).entity(result);
             BrowserSecurityHeaderSetup.headers(builder, realm);
-            LocaleHelper.updateLocaleCookie(builder, locale, realm, uriInfo, Urls.localeCookiePath(baseUri,realm.getName()));
+
+            String keycloakLocaleCookiePath = Urls.localeCookiePath(baseUri, realm.getName());
```

Comment:
I've got it working with $translate.use.  The only caveat is that the KEYCLOAK_LOCALE cookie needs to use root as its path.  The reason is because to share a cookie both angular and Freemarker pages need a common subpath.  The only one guaranteed to work for both is "/".  Really, that's been the sticking point all along.

I don't know of any reason why using the root path for a cookie would cause a problem, but I wanted to make you aware in case you know of problems with that.  If you are OK with this solution I'll clean everything up and update the PR.